pkg/infrastructure/ibmcloud/clusterapi/clusterapi.go

```diff
@@ -0,0 +1,25 @@
+package clusterapi
+
+import (
+	"context"
+
+	"github.com/openshift/installer/pkg/infrastructure/clusterapi"
+	ibmcloudtypes "github.com/openshift/installer/pkg/types/ibmcloud"
+)
+
+// Provider is the IBM Cloud implementation of the clusterapi InfraProvider.
+type Provider struct {
+	clusterapi.InfraProvider
+}
+
+var _ clusterapi.PreProvider = Provider{}
```

Comment:
In case we change the `Provider` interface in the future:
```suggestion
var _ clusterapi.Provider = Provider{}
var _ clusterapi.PreProvider = Provider{}
```

Reply:
Updated, I dropped `PreProvider` in favor of just `Provider`.

Reply:
But I see you still have an empty `PreProvision` function. If you intend on implementing that, I don't see an issue with keeping also the guard for the `PreProvider` interface.

Reply:
I suggest to either:
1. remove the `PreProvision` function since it's doing nothing. Later on, when you implement it, add also the `var _ clusterapi.PreProvider = Provider{}` guard.
2. Leave the empty `PreProvision` function and add the `PreProvider` guard since you intend on implementing the functionality in follow-up PRs.

Reply:
Updated.
Okay, I believe we need a PreProvision, to setup the RHCOS image, so I will expect to add that functionality to PreProvision later, and re-incorporated the PreProvider guard.